torchaudio/datasets/tedlium.py

```diff
@@ -0,0 +1,159 @@
+        """
+        self._ext_audio = audio_ext
+        if release in _RELEASE_CONFIGS.keys():
+            folder_in_archive = _RELEASE_CONFIGS[release]["folder_in_archive"]
+            url = _RELEASE_CONFIGS[release]["url"]
+            subset = subset if subset else _RELEASE_CONFIGS[release]["subset"]
+        else:
+            # Raise warning
+            raise RuntimeError(
+                "The release {} does not match any of the supported tedlium releases{} ".format(
+                    release, _RELEASE_CONFIGS.keys(),
+                )
+            )
+
+        basename = os.path.basename(url)
+        archive = os.path.join(root, basename)
+
+        basename = basename.split(".")[0]
+
+        self._path = os.path.join(root, folder_in_archive, _RELEASE_CONFIGS[release]["data_path"])
+        if subset in ["train", "dev", "test"]:
+            self._path = os.path.join(self._path, subset)
+        if download:
+            if not os.path.isdir(self._path):
+                if not os.path.isfile(archive):
+                    checksum = _RELEASE_CONFIGS[release]["checksum"]
+                    download_url(url, root, hash_value=checksum)
+                extract_archive(archive)
+
+        self._walker = []
+        for file in walk_files(self._path, suffix=".stm", prefix=False, remove_suffix=True):
+            stm_path = os.path.join(self._path, "stm", file + ".stm")
+            with open(stm_path) as f:
+                l = len(f.readlines())
+                self._walker.extend((file, line) for line in range(l))
+
+    def load_tedlium_item(self, fileid: str, line: int, path: str) -> Tedlium_item:
+        """Loads a TEDLIUM dataset sample given a file name and corresponding sentence name
+
+        Args:
+            fileid (str): File id to identify both text and audio files corresponding to the sample
+            line (int): Line identifier for the sample inside the text file
+            path (str): Dataset root path
+
+        Returns:
+            Tedlium_item: A namedTuple containing [waveform, sample_rate, transcript, talk_id, speaker_id, identifier]
+        """
+        transcript_path = os.path.join(path, "stm", fileid)
+        with open(transcript_path + ".stm") as f:
+            transcript = f.readlines()[line]
+            talk_id, _, speaker_id, start_time, end_time, identifier, transcript = transcript.split(" ", 6)
+
+        wave_path = os.path.join(path, "sph", fileid)
+        # Calculate indexes for start time and endtime
+        start_time = int(float(start_time) * 16000)
+        end_time = int(float(end_time) * 16000)
+        waveform, sample_rate = self.load_audio(
+            wave_path + self._ext_audio, frame_offset=start_time, num_frames=end_time - start_time
+        )
+        return Tedlium_item(waveform, sample_rate, transcript, talk_id, speaker_id, identifier)
+
+    def load_audio(self, path: str, frame_offset: int = 0, num_frames: int = -1) -> [Tensor, int]:
```

Comment:
I would do

`def load_audio(self, path:str, start_time:float, end_time:float, sample_rate:int = 16000)`

that way, we can delegate the responsibility of frame calculation into `load_audio` and `load_tedlium_item`'s responsibility becomes more focused on finding the right item/path.

Reply:
Also could you prefix the function name with underscore `_load_audio` to let users know it's not for public use?

Reply:
done in the last commit, :+1: